frontend/src/pages/modelServing/screens/global/ServeModelButton.tsx

```diff
@@ -58,7 +59,7 @@ const ServeModelButton: React.FC = () => {
           getProjectModelServingPlatform(project, servingPlatformStatuses).platform,
         )
       }
-      isAriaDisabled={!project || !templatesEnabled}
+      isAriaDisabled={!project || (!isNIMAvailable && isKServeNIMEnabled)}
```

Comment:
```suggestion
      isAriaDisabled={!project || !templatesEnabled || (!isNIMAvailable && isKServeNIMEnabled)}
```

We should probably preserve the existing behaviour for if there are no serving runtime templates. (If there are none, deployments are not possible)

Reply:
@emilys314 The account.status is our final source of truth because it is only set to true when all five conditions are met. While the template may be enabled, NIM is not considered available if any of the other conditions are not true.

The conditions that must be true are:

APIKeyValidation
ConfigMapUpdate
TemplateUpdate
SecretUpdate
AccountStatus
This ensures that deployments are only possible when the entire setup is in a valid state.

Reply:
Thanks for the info. Although I think if this is not a nim project, the error can still occur.

So NIM takes it into account, but not kserve. so here i think we should keep it. Your right on the other comment though

Reply:
Updated